packages/agent/src/certificate.ts

```diff
@@ -102,20 +102,19 @@ function isBufferEqual(a: ArrayBuffer, b: ArrayBuffer): boolean {
 export class Certificate {
   private readonly cert: Cert;
   private verified = false;
-  private _rootKey: ArrayBuffer | null = null;
 
-  constructor(response: ReadStateResponse, private _agent: Agent = getDefaultAgent()) {
-    this.cert = cbor.decode(new Uint8Array(response.certificate));
+  constructor(certificate: ArrayBuffer, private _rootKey: Promise<ArrayBuffer>) {
```

Comment:
If we change the interface to stick the `canisterId` into the certificate that makes total sense - if we for whatever reason want to call `verify` separately (I don't know why we would), a `Promise` would allow to delay blocking until we need it.